File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 10/27/2025, 7:10:43 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) {
        map<int,int>mp;
        int i,more,cur,j;
        for(i=0;i<arr.size();i++){
            cur=arr[i];
            more=target-cur;
            if(mp.find(more)!=mp.end()){
                return {i,mp[more]};
            }
            else{
                mp[cur]=i;
            }
        }
        return {};
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 10/27/2025, 7:10:42 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_mp={}
        for i,num in enumerate(nums):
            diff=target-num
            if diff in num_mp:
                return [num_mp[diff],i]
            num_mp[num]=i